pallets/xyk-liquidity-mining/benchmarking/src/lib.rs

```diff
@@ -0,0 +1,461 @@
+		let xyk_caller = create_funded_account::<T>("xyk_caller", 1);
+		let liq_provider = create_funded_account::<T>("liq_provider", 2);
+
+		initialize_pool::<T>(xyk_caller, ASSET_PAIR.asset_in, ASSET_PAIR.asset_out, 1_000_000 * ONE, Price::from(10))?;
+		let xyk_id = xykpool::Pallet::<T>::pair_account_from_assets(ASSET_PAIR.asset_in, ASSET_PAIR.asset_out);
+		xyk_add_liquidity::<T>(liq_provider.clone(), ASSET_PAIR, 100_000, 1_000_000_000)?;
+
+		lm_create_global_farm::<T>(1_000_000 * ONE, caller.clone(), Perquintill::from_percent(20))?;
+		lm_create_yield_farm::<T>(caller, GLOBAL_FARM_ID, ASSET_PAIR, FixedU128::one())?;
+
+		lm_deposit_shares::<T>(liq_provider.clone(), ASSET_PAIR, 10_000)?;
+		set_block_number::<T>(100_000);
+	}: {
+		LiquidityMining::<T>::deposit_shares(RawOrigin::Signed(liq_provider.clone()).into(), GLOBAL_FARM_ID, YIELD_FARM_ID, ASSET_PAIR, 10_000)?
+	}
+
+	redeposit_lp_shares {
+		let caller = create_funded_account::<T>("caller", 0);
+		let xyk_caller = create_funded_account::<T>("xyk_caller", 1);
+		let liq_provider = create_funded_account::<T>("liq_provider", 2);
+		let shares_amount = 10_000;
+
+		initialize_pool::<T>(xyk_caller, ASSET_PAIR.asset_in, ASSET_PAIR.asset_out, 1_000_000 * ONE, Price::from(10))?;
+		xyk_add_liquidity::<T>(liq_provider.clone(), ASSET_PAIR, shares_amount, 1_000_000_000)?;
+
+		//global id: 1, yield id: 2
+		lm_create_global_farm::<T>(100_000 * ONE, caller.clone(), Perquintill::from_percent(20))?;
+		lm_create_yield_farm::<T>(caller.clone(), GLOBAL_FARM_ID, ASSET_PAIR, FixedU128::one())?;
+
+		//global id: 3, yield id: 4
+		lm_create_global_farm::<T>(100_000 * ONE, caller.clone(), Perquintill::from_percent(20))?;
+		lm_create_yield_farm::<T>(caller.clone(), GLOBAL_FARM_ID_2, ASSET_PAIR, FixedU128::one())?;
+
+		//global id: 5, yield id:6
+		lm_create_global_farm::<T>(100_000 * ONE, caller.clone(), Perquintill::from_percent(20))?;
+		lm_create_yield_farm::<T>(caller.clone(), 5, ASSET_PAIR, FixedU128::one())?;
+
+		//global id: 7, yield id:8
+		lm_create_global_farm::<T>(100_000 * ONE, caller.clone(), Perquintill::from_percent(20))?;
+		lm_create_yield_farm::<T>(caller.clone(), 7, ASSET_PAIR, FixedU128::one())?;
+
+		//global id: 9, yield id:10
+		lm_create_global_farm::<T>(100_000 * ONE, caller.clone(), Perquintill::from_percent(20))?;
+		lm_create_yield_farm::<T>(caller, 9, ASSET_PAIR, FixedU128::one())?;
+
+		set_block_number::<T>(200_000);
+
+		LiquidityMining::<T>::deposit_shares(RawOrigin::Signed(liq_provider.clone()).into(), GLOBAL_FARM_ID, YIELD_FARM_ID, ASSET_PAIR, shares_amount)?;
+		LiquidityMining::<T>::redeposit_lp_shares(RawOrigin::Signed(liq_provider.clone()).into(), GLOBAL_FARM_ID_2, YIELD_FARM_ID_2, ASSET_PAIR, DEPOSIT_ID)?;
+		LiquidityMining::<T>::redeposit_lp_shares(RawOrigin::Signed(liq_provider.clone()).into(), 5, 6, ASSET_PAIR, DEPOSIT_ID)?;
+		LiquidityMining::<T>::redeposit_lp_shares(RawOrigin::Signed(liq_provider.clone()).into(), 7, 8, ASSET_PAIR, DEPOSIT_ID)?;
```

Comment:
I'll double check but I think there was a difference when there was only 1 entry in the deposit. I'll add comment.
In the regular deposit it's not necessary because it creates new deposit